Sources/Sentry/SentryPerformanceTracker.m

```diff
@@ -79,15 +78,13 @@ - (SentrySpanId *)startSpanWithName:(NSString *)name
             }
 
             SENTRY_LOG_DEBUG(@"Creating new transaction bound to scope: %d", bindToScope);
-            newSpan = [SentrySDK.currentHub startTransactionWithContext:context
-                                                            bindToScope:bindToScope
-                                                        waitForChildren:YES
-                                                  customSamplingContext:@{}
-                                                           timerWrapper:nil];
-
-            if ([newSpan isKindOfClass:[SentryTracer class]]) {
-                [(SentryTracer *)newSpan setDelegate:self];
-            }
+            SentryTracer *newTracer = [SentrySDK.currentHub startTransactionWithContext:context
+                                                                            bindToScope:bindToScope
+                                                                        waitForChildren:YES
+                                                                  customSamplingContext:@{}
+                                                                           timerWrapper:nil];
+            newSpan = newTracer.rootSpan;
+            [newTracer setDelegate:self];
```

Comment:
Here's another place that is much cleaner. We shouldn't need to check the underlying type of an object and cast downwards. Also, what happens if the `if` condition were false?

Reply:
I guess this makes sense. startTransaction always return SentryTracer.